thirdparty/CMakeLists.txt

```diff
@@ -97,19 +97,19 @@ cmake_dependent_option(ADIOS2_USE_EXTERNAL_EVPATH
 )
 cmake_dependent_option(ADIOS2_USE_EXTERNAL_ATL
   "Use an externally supplied ATL library" OFF
-  "NOT ADIOS2_USE_EXTERNAL_EVPATH" OFF
```

Comment:
This means that ADIOS2_USE_EXTERNAL_ATL will not be internal (user can set it) when ADIOS2_USE_EXTERNAL_EVPATH=OFF. You probably mean the opposite which is use can set ADIOS2_USE_EXTERNAL_ATL when ADIOS2_USE_EXTERNAL_EVPATH=ON

https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html